torchmultimodal/modules/encoders/mdetr_text_encoder.py

```diff
@@ -0,0 +1,284 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Union
+
+import torch
+from torch import nn, Tensor
+from torchtext.models import ROBERTA_BASE_ENCODER
+
+
+def create_position_ids_from_input_ids(input_ids: Tensor, padding_idx: int):
+    """
+    Replace non-padding symbols with their position numbers.
+    Position numbers begin at padding_idx+1. Padding symbols
+    are ignored. This is modified from fairseq's `utils.make_positions`.
+
+    Inputs:   input_ids (torch.Tensor): Tensor from which to create position IDs.
+              padding_idx (int): Padding index
+                (determines starting point of position IDs).
+    """
+    mask = input_ids.ne(padding_idx).int()
+    incremental_indices = torch.cumsum(mask, dim=1).type_as(mask) * mask
+    return incremental_indices.long() + padding_idx
+
+
+class MDETRTextEmbeddings(nn.Module):
+    """
+    Class for RoBERTa embeddings as used in MDETR. This is very similar to the FLAVA
```

Comment:
yeah idk this should be same as flava in theory (both roberta?). are you alluding to the handling of mask in position embeddings?

Reply:
I thought FLAVA was using BERT embeddings? At least based on [this comment](https://github.com/facebookresearch/multimodal/blob/82480d8cf4a9359ecc28fbb4de257785bf25a1c3/torchmultimodal/models/flava/flava_text_encoder.py#L23)

Reply:
Thanks, lemme remove the reference to BERT. I still think the handling of position IDs is different though (at least when they are not passed, which is the case for us). In FLAVA they are just taking a [slice](https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/models/flava/flava_text_encoder.py#L73-L75), whereas here we explicitly use the `padding_idx`